src/kudu/consensus/raft_consensus.h

```diff
@@ -1267,6 +1272,9 @@ class RaftConsensus : public std::enable_shared_from_this<RaftConsensus>,
   scoped_refptr<Counter> raft_proxy_num_requests_log_read_timeout_;
   scoped_refptr<Counter> raft_proxy_num_requests_hops_remaining_exhausted_;
 
+  CompressionCodec* codec_ = nullptr;
```

Comment:
Any reason not to preserve std::atomic<const CompressionCodec*> codec_ 
like LogCache used to be?

Reply:
It's protected by RaftConsensus::lock_